asllib/ASLSemanticsReference.mld

```diff
@@ -321,47 +324,78 @@ Evaluation of Left-Hand-Side Expressions
 }
 [eval_lexpr version env le m] is [env[le --> m]]. 
 
-The evaluation of the expression [le] under an environment [env]
-is [v] and one of the following applies:
+The evaluation of the left-hand-side expression [le] associated with a value
+[v] under an environment [env] is [new_env] and one of the following applies:
 
   {3 Rule.LEIgnore}
-  <description> 
+  [le] can be ignored and [new_env] is [env]. 
+
   <minimal example>
 
   {3 Rule.LELocalVar}
-  <description> 
+  [le] denotes a variable [x] and all of the following applies:
+  - [new_env] is a local environment; 
+  - [new_env] is [env] where [x] has been assigned the value [v].
+
   <minimal example>
 
   {3 Rule.LEGlobalVar}
-  <description> 
-  <minimal example>
+  [le] denotes a variable [x] and all of the following applies:
+  - [new_env] is a global environment; 
+  - [new_env] is [env] where [x] has been assigned the value [v].
 
-  {3 Rule.LEUndefIdentV0}
-  <description> 
   <minimal example>
 
   {3 Rule.LEUndefIdentV1}
-  <description> 
+  [le] denotes a variable [x] which is not bound in [env] and all of the following applies:
+  - the ASL language version is V1;
+  - an Error "UndefinedIdentifier" is raised.
+
+  <minimal example>
+
+  {3 Rule.LEUndefIdentV0}
+  [le] denotes a variable [x] which is not bound in [env] and all of the following applies:
+  - the ASL language version is V0;
+  - [new_env] is [env] where [x] has been declared a local identifier bound to the value [v].
+
   <minimal example>
 
   {3 Rule.LESlice}
-  <description> 
+  [le] denotes an expression [e_bv] sliced as per [slices] and all of the following applies:
+  - The evaluation of [e_bv] under [env] is [v_bv];
+  - The evaluation of [slices] under [env] is [positions];
+  - [v] is the value read in [v_bv] from [positions];
+  - [new_env] is [env] where [le] has been assigned the value [v].
```

Comment:
This is beginning to be tricky. Here is a suggestion, but I'm not sure that it is understandable:

```suggestion
  [le] denotes a left-hand-side expression [le_bv] sliced as per [slices] and all of the following applies:
  - [le_bv] has a corresponding right-hand-side expression [e_bv];
  - The evaluation of [e_bv] under [env] is [v_bv];
  - The evaluation of [slices] under [env] is [positions];
  - [new_v_bv] is [v_bv] where the positions [positions] have been updated to [v]
  - [new_env] is [env] where [le_bv] has been assigned the value [new_v_bv].
```